src/types/sgrStack.cpp

```diff
@@ -0,0 +1,240 @@
+
+        TextAttribute result = currentAttributes;
+
+        // From xterm documentation:
+        //
+        //  CSI # {
+        //  CSI Ps ; Ps # {
+        //            Push video attributes onto stack (XTPUSHSGR), xterm.  The
+        //            optional parameters correspond to the SGR encoding for video
+        //            attributes, except for colors (which do not have a unique SGR
+        //            code):
+        //              Ps = 1  -> Bold.
+        //              Ps = 2  -> Faint.
+        //              Ps = 3  -> Italicized.
+        //              Ps = 4  -> Underlined.
+        //              Ps = 5  -> Blink.
+        //              Ps = 7  -> Inverse.
+        //              Ps = 8  -> Invisible.
+        //              Ps = 9  -> Crossed-out characters.
+        //              Ps = 2 1  -> Doubly-underlined.
+        //              Ps = 3 0  -> Foreground color.
+        //              Ps = 3 1  -> Background color.
+        //
+        //  (some closing braces for people with editors that get thrown off without them: }})
+        //
+        // Note that not all of these attributes are actually supported by
+        // renderers/conhost, despite setters/getters on TextAttribute.
+
+        try
+        {
+            // Boldness = 1,
+            if (validParts.test(static_cast<size_t>(SgrSaveRestoreStackOptions::Boldness)))
+            {
+                result.SetBold(savedAttribute.IsBold());
+            }
+
+            // Faintness = 2,
+            if (validParts.test(static_cast<size_t>(SgrSaveRestoreStackOptions::Faintness)))
+            {
+                result.SetFaint(savedAttribute.IsFaint());
+            }
+
+            // Italics = 3,
+            if (validParts.test(static_cast<size_t>(SgrSaveRestoreStackOptions::Italics)))
+            {
+                result.SetItalic(savedAttribute.IsItalic());
+            }
+
+            // Underline = 4,
+            if (validParts.test(static_cast<size_t>(SgrSaveRestoreStackOptions::Underline)))
+            {
+                result.SetUnderlined(savedAttribute.IsUnderlined());
+            }
+
+            // Blink = 5,
+            if (validParts.test(static_cast<size_t>(SgrSaveRestoreStackOptions::Blink)))
+            {
+                result.SetBlinking(savedAttribute.IsBlinking());
+            }
+
+            // Negative = 7,
+            if (validParts.test(static_cast<size_t>(SgrSaveRestoreStackOptions::Negative)))
+            {
+                if (savedAttribute.IsReverseVideo())
+                {
+                    if (!result.IsReverseVideo())
+                    {
+                        result.Invert();
+                    }
+                }
+                else
+                {
+                    if (result.IsReverseVideo())
+                    {
+                        result.Invert();
+                    }
+                }

```

Comment:
wow, i wish we had a "SetReverseVideo".

Reply:
We do have a `SetReverseVideo` (although it may not have existed when this PR was first written).